internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -191,3 +181,61 @@ func (c *CodePipeline) getPipelineName(resourceArn string) (string, error) {
 
 	return parsedArn.Resource, nil
 }
+
+// GetPipelineStatus retrieves status information from a given pipeline.
+func (c *CodePipeline) GetPipelineState(name string) (*PipelineState, error) {
+	input := &cp.GetPipelineStateInput{
+		Name: aws.String(name),
+	}
+	resp, err := c.client.GetPipelineState(input)
+
+	if err != nil {
+		return nil, fmt.Errorf("get pipeline state %s: %w", name, err)
+	}
+	var stageStates []*StageState
+	for _, stage := range resp.StageStates {
+		var stageName string
+		if stage.StageName != nil {
+			stageName = aws.StringValue(stage.StageName)
+		}
+		var transition string
+		if stage.InboundTransitionState != nil {
+			transition = "DISABLED"
+			if *stage.InboundTransitionState.Enabled == true {
+				transition = "ENABLED"
+			}
+		}
+		var status string
+		for _, actionState := range stage.ActionStates {
+			if actionState.LatestExecution != nil {
+				status = aws.StringValue(actionState.LatestExecution.Status)
+			}
+		}
```

Comment:
That's correct. There can be several actions in one stage, each with different statuses. Efe, I like your suggestion - maybe we can add a method to StageState struct which can compute an aggregate stage status based on the actions. 

The algorithm that the console uses to do this is:
* Stage state is in progress if any action in that stage is in progress
* Stage state is failed if any action in that stage is failed or abandoned (and none are in progress)
* Stage state is succeeded if all actions are succeeded
* Otherwise the stage state is in unknown

Does that make sense?

Reply:
Ahh, thanks for catching this! I should have looked beyond my example in the API reference that has only one action. 

@kohidave If stage state is unknown, the field is not present, right?

Reply:
@efekarakus How should I display these in HumanString output?

Reply:
I think David's recommendation is great, we can add a method to a `StageState` object like:
```go
func (ss StageState) AggregateStatus() string {
    // Loops through ss.Actions and does the algorithm David described above.
}
```
and then in:
```go
func (ss StageState) HumanString() string {
    return fmt.Sprintf("  %s\t%s\t%s\n", ss.StageName, ss.AggregateStatus(), ss.Transition) 
}
```

Reply:
:+1:  to the above. Here's an example of what multiple actions in a stage  looks like in the codepipeline console:
<img width="737" alt="codepipeline_console" src="https://user-images.githubusercontent.com/29964746/83290623-c0827780-a19b-11ea-956a-3e0692e68d2a.png">